src/test/java/jep/test/TestPyBuiltins.java

```diff
@@ -0,0 +1,336 @@
+            return false;
+        }
+        return true;
+    }
+
+    private boolean testHasAttr() {
+        if (!builtins.hasattr(object, "__str__")){
+            failure = "hasattr builtin does not find __str__";
+            return false;
+        }
+        if (builtins.hasattr(object, "fakeAttribute")){
+            failure = "hasattr builtin found fakeAttribute";
+            return false;
+        }
+        return true;
+    }
+
+    private boolean testId() {
+        if (builtins.id(object) != builtins.id(object)){
+            failure = "id builtin returns inconsistent id";
+            return false;
+        }
+        if (builtins.id(object) == builtins.id(objectType)){
+            failure = "id builtin returns equal id for inequal objects";
+            return false;
+        }
+        return true;
+    }
+
+    private boolean testIsInstance() {
+        if (!builtins.isinstance(object, objectType)){
+            failure = "isinstance builtin returns an object is not an object type";
+            return false;
+        }
+        if (builtins.isinstance(object, subtypeOne)){
+            failure = "isinstance builtin returns an object is a subtype";
+            return false;
+        }
+        return true;
+    }
+
+    private boolean testIsSubClass() {
+        if (!builtins.issubclass(subtypeOne, objectType)){
+            failure = "issubclass builtin returns subtype is not a subclass of object";
+            return false;
+        }
+        if (builtins.isinstance(subtypeOne, subtypeTwo)){
+            failure = "issubclass builtin returns unrelated subtype is a subclass";
+            return false;
+        }
+        return true;
+    }
+
+    private boolean testList() {
+        PyObject listBuiltin = builtins.list();
+        PyObject listGetValue = interp.getValue("[]", PyObject.class);
+        if (!listBuiltin.equals(listGetValue)){
+            failure = "list builtin does not return an empty list";
+            return false;
+        }
+        listBuiltin = builtins.list(new String[] {"1", "2", "3"});
+        listGetValue = interp.getValue("['1','2','3']", PyObject.class);
+        if (!listBuiltin.equals(listGetValue)){
+            failure = "list builtin does not return a full list";
+            return false;
+        }
+        return true;
+    }
+
+    private boolean testObject() {
+        PyObject o = builtins.object();
+        if (!o.equals(object)){
+            failure = "object builtin does not return an empty object";
+            return false;
+        }
+        return true;
+    }
+
+    private boolean testSet() {
+        PyObject setBuiltin = builtins.set();
+        PyObject setGetValue = interp.getValue("set()", PyObject.class);
+        if (!setBuiltin.equals(setGetValue)){
+            failure = "set builtin does not return an empty set";
+            return false;
+        }
+        setBuiltin = builtins.set(new String[] {"1", "2", "3"});
```

Comment:
Not a big deal, but a bit surprising that you support list(arg) and set(arg) but not dict(args).

Reply:
I just think it would confuse java developers to try to figure out what types of things to pass as an arg to a dict. I think it would have top be an iterable containing iterables of length 2 so like a List<List<?>> or an Object[][]. It seems like anyone who needs that and understands what should go into it would not be afraid to implement their own more specific proxy.